### PR TITLE
custom logs support

### PR DESCRIPTION
**Pull request by simathih**

### CODE REVIEW

File: AzureMonitorAgent/agent.py [hunk 1]
```diff
@@ -256,6 +257,47 @@ def check_kill_process(pstring):
...
+                # Removing the file in case it is already being run in a process,
+                # in which case we can get an error "text file busy" while copying
+                os.remove(pa_bin)
+                copyfile(pa_bin_local_path, pa_bin)
+                os.chmod(pa_bin, stat.S_IXGRP | stat.S_IRGRP | stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR | stat.S_IXOTH | stat.S_IROTH)
+
+        else:
+            # No previous binary exist, simply copy it and make it executable
+            copyfile(pa_bin_local_path, pa_bin)
+            os.chmod(pa_bin, stat.S_IXGRP | stat.S_IRGRP | stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR | stat.S_IXOTH | stat.S_IROTH)
```
Comment (canfikret) [inline AzureMonitorAgent/agent.py:279]: Lines 274 and 279 are repeated. You can have only one outside of conditional statements.

File: AzureMonitorAgent/agent.py [hunk 2]
```diff
@@ -256,6 +257,47 @@ def check_kill_process(pstring):
...
+                  
+    agentlauncher_bin_local_path = os.getcwd() + "/agentLauncherBin/agentlauncher"
+    agentlauncher_bin = "/opt/microsoft/azuremonitoragent/bin/agentlauncher"
+
+    # Check if previous file exist at the location, compare the two binaries,
```
Comment (canfikret) [inline AzureMonitorAgent/agent.py:284]: This can be a method on its own. CompareAndCopyBin(src, dest) ?

**Review by canfikret — CHANGES_REQUESTED** [review]

File: AzureMonitorAgent/agent.py [hunk 3]
```diff
@@ -722,10 +778,26 @@ def metrics_watcher(hutil_error, hutil_log):
...
+                data = f.read()
+
+                if (data != ''):
+                    crc_fluent = hashlib.sha256(data.encode('utf-8')).hexdigest()
+
+                    if (crc_fluent != last_crc_fluent):
+                        restart_launcher()
+                        last_crc_fluent = crc_fluent
+            else:
+                if last_crc_fluent != None:
```
Comment (simathih) [inline AzureMonitorAgent/agent.py:797]: remove

File: AzureMonitorAgent/agent.py [hunk 4]
```diff
@@ -384,6 +417,8 @@ def enable():
         return 0, ""
     elif (protected_settings is None or len(protected_settings) == 0) or (public_settings is not None and "proxy" in public_settings and "mode" in public_settings.get("proxy") and public_settings.get("proxy").get("mode") == "application"):
         default_configs["ENABLE_MCS"] = "true"
+        default_configs["PA_GIG_BRIDGE_MODE"] = "true"
+        default_configs["PA_FLUENT_SOCKET_PORT"] = "13000"
```
Comment (canfikret) [inline AzureMonitorAgent/agent.py:421]: Put a note that this port will be provided dynamically later on.

File: AzureMonitorAgent/agent.py [hunk 5]
```diff
@@ -722,10 +778,22 @@ def metrics_watcher(hutil_error, hutil_log):
     # Sleep before starting the monitoring
     time.sleep(sleepTime)
     last_crc = None
+    last_crc_fluent = None
     me_msi_token_expiry_epoch = None
 
     while True:
         try:
+            if os.path.isfile(FluentCfgPath):
```
Comment (canfikret) [inline AzureMonitorAgent/agent.py:787]: Take a note that it is the agentlauncher's responsibility to detect config file changes. Kennedy is going to add that in next iteration.

**Review by canfikret — APPROVED** [review]

File: AzureMonitorAgent/agent.py (hunk 5 above)
Comment (simathih) [reply AzureMonitorAgent/agent.py:787]: yeah, we will remove this then

**Pull request merged**